tools/Makefile.shared

```diff
@@ -12,10 +12,62 @@
 #*   special exception on linking described in the file LICENSE.          *
 #*                                                                        *
 #**************************************************************************
-
+MAKEFLAGS := -r -R
```

Comment:
What's the point? You're not doing any recursive `make` invocation in this directory.